lib/promscrape/scraper.go

```diff
@@ -64,7 +67,7 @@ func runScraper(configFile string, pushData func(wr *prompbmarshal.WriteRequest)
 	signal.Notify(sighupCh, syscall.SIGHUP)
 
 	logger.Infof("reading Prometheus configs from %q", configFile)
-	cfg, data, err := loadConfig(configFile)
+	cfg, _, err := loadConfig(configFile)
```

Comment:
as all sws can be restarted smarter, check whether configFile change is not necessary now